eventstore/tasks.py

```diff
@@ -532,3 +538,75 @@ def process_ada_assessment_notification(
     triage.risk = triage.calculate_risk()
     triage.full_clean()
     triage.save()
+
+
+@app.task(
+    autoretry_for=(RequestException, SoftTimeLimitExceeded, TembaHttpError),
+    retry_backoff=True,
+    max_retries=15,
+    acks_late=True,
+    soft_time_limit=10,
+    time_limit=15,
+)
+def post_random_contacts_to_slack_channel():
+    # Get 10 random contacts to post to slack channel
+    if settings.RAPIDPRO_URL and settings.RAPIDPRO_TOKEN:
+        rapidpro_url = urljoin(settings.RAPIDPRO_URL, "/contact/read/{}/")
+
+        contact_details = []
+
+        while len(contact_details) < settings.RANDOM_CONTACT_LIMIT:
+            profile_link, contact_uuid = get_random_contact()
+
+            rapidpro_link = rapidpro_url.format(contact_uuid)
+            links = {"Rapid Pro Link: ": rapidpro_link, "Turn Link": profile_link}
+
+            contact_details.append(links)
+
+        if contact_details:
+            sent = send_slack_message("test-mon", str(contact_details))
+            return {"success": sent, "results": contact_details}
+        return {"success": False, "results": contact_details}
+
+
+def get_turn_profile_link(contact_number):
+    if settings.TURN_URL and settings.TURN_TOKEN:
+        turn_header = {
+            "Authorization": "Bearer {}".format(settings.TURN_TOKEN),
+            "Accept": "application/vnd.v1+json",
+        }
+
+        if contact_number:
+            turn_url = settings.TURN_URL + "/v1/contacts/{}/messages".format(
+                contact_number
+            )
+
+            profile = requests.get(url=turn_url, headers=turn_header)
+
+            if profile:
+                # Get turn message link
+                return profile.json().get("chat").get("permalink")
```

Comment:
This will fail if the `chat` key doesn't exist, you can return {} as the default in the first `.get()`